pkg/kubectl/generate.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/spf13/cobra"
+)
+
+// GeneratorParam is a parameter for a generator
+type GeneratorParam struct {
+	Name     string
```

Comment:
Add a TODO: facilitate structured json generator input schemas.

This approach would force generators that accepted nested or repeated entities to flatten their inputs rather than just accept json, which is my preferred approach.

However, since this isn't exposed to users, it isn't critical to fix in this PR.